src/nfa/nfa.cc

```diff
@@ -1419,6 +1411,26 @@ Nfa Mata::Nfa::construct(
         aut.delta.add(src_state, symbol, tgt_state);
     }
 
+    std::unordered_set<std::string> final_formula_nodes;
+    if (!(inter_aut.final_formula.node.name == "true" || inter_aut.final_formula.node.name == "false")) { 
+        final_formula_nodes = inter_aut.final_formula.collect_node_names();
+    }
+    bool final_nodes_are_negated = (inter_aut.final_formula.node.name == "true" || inter_aut.are_final_states_conjunction_of_negation());
```

Comment:
We should discuss whether this should not be a function in inter_automaton, something like `inter_aut.final_nodes_are_negated()`. The same for condition above.

Reply:
I don't think this should be a function of inter_automaton, I am asking if I have conjunction of negations or the constant true, function call would just make it less obvious.
The condition `inter_aut.final_formula.node.name == "true" || inter_aut.final_formula.node.name == "false"` should probably be a function of inter_automaton, or better, function of FormulaGraph, something like `is_constant`, I'll add it. A cleaner solution would maybe be adding constants to FormulaGraph.

Reply:
Yes, it would be better to modify Node and add an option to enum that the node is a constant operand true/false.

Reply:
I added an issue #174 for this